File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
      numbers = {}
      
      for i in range(len(nums)):
        numbers[nums[i]] = i
        
      for i in range(len(nums)):
        if numbers.get(target - nums[i]) and i != numbers[target - nums[i]]: 
          return [numbers[target - nums[i]], i]
            
      return [0, 1]

File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
      
      map<int, int> outputs;
      
      for (int i = 0; i < nums.size(); i++) {
        auto res = outputs.find(nums[i]);
        if (res != outputs.end()) {
          return {i, res->second};
        } else {
          outputs.insert(make_pair(target-nums[i], i));
        }
      }
      return {-1, -1};
    }
};